Standards/scs-XXXX-v1-default-rules-for-security-groups.md

```diff
@@ -0,0 +1,95 @@
+
+The Whitelisting concept of Security Group rules makes it hard to allow traffic with an exception of certain ports.
+It would be possible to just define many rules to achieve what a blacklist would achieve.
+This has the severe downside that users could be confused by these rules and will not disable unnecessary default rules in their SGs.
+
+## Standard
+
+The default Security Group rules for ALL Security Groups MUST NOT allow incoming Traffic. Neither IPv4 nor IPv6.
+This can be achieved through the absence of any ingress rules in the default Security Group rules.
+
+The default Security Group rules for ALL Security Groups SHOULD allow egress Traffic for both IPv4 and IPv6.
+This standard should not forbid to also disallow all outgoing traffic.
+Allowing all outgoing traffic in the default rules in combination with blocking all incoming traffic would be strict enough from a security point of view.
+And it would make it necessary for users to check and change the rules of their security group to a meaningful set.
+
+### Example
+
+In the following table, there is no ingress traffic allowed but all egress traffic:
+
+```bash
+$ openstack default security group rule list
++--------------------------+-------------+-----------+-----------+------------+-----------+-----------------------+----------------------+--------------------------------+-------------------------------+
+| ID                       | IP Protocol | Ethertype | IP Range  | Port Range | Direction | Remote Security Group | Remote Address Group | Used in default Security Group | Used in custom Security Group |
++--------------------------+-------------+-----------+-----------+------------+-----------+-----------------------+----------------------+--------------------------------+-------------------------------+
+| 47b929fd-9b39-4f22-abc5- | None        | IPv6      | ::/0      |            | egress    | None                  | None                 | True                           | True                          |
+| 7d4f64d10909             |             |           |           |            |           |                       |                      |                                |                               |
+| 92a79600-5358-4fef-a390- | None        | IPv4      | 0.0.0.0/0 |            | egress    | None                  | None                 | True                           | True                          |
+| 822665f46070             |             |           |           |            |           |                       |                      |                                |                               |
++--------------------------+-------------+-----------+-----------+------------+-----------+-----------------------+----------------------+--------------------------------+-------------------------------+
+```
+
+These rules can also be configured to only apply to custom Security Groups through the API.
```

Comment:
Now that I think about it, I'm wondering if instead of quoting an arbitrary sample output of `openstack default security group rule create ... --for-custom-sg` we should rather provide more relevant commands to help CSPs to enforce the standard.

I think the following contents would make sense in the example section:

1. How to make sure that no ingress rules are defined for both custom and default SGs.
    - (The list output table above already does this, maybe just add a sentence about what to look out for, e.g. no rows with ingress in them etc.)
2. If any conflicting ingress rules exist, how to remove them using CLI commands.
3. For the optional egress rules just add a small hint that there is `--for-custom-sg` and `--for-default-sg` differentiation when using the CLI but do not include example outputs for the `create` command like below.

Reply:
Maybe that is a little bit too much for the standard and would better fit in a guide?
In the first version i just wanted to show, how the default SG rules table can be looked up, and how it is supposed to look.

Reply:
Fair point. I think we should only keep the table above then and state that this is how it is supposed to look like with no conflicting ingress rules defined.

In turn, I think the example below about `--for-custom-sg` should be removed because it paints an incomplete picture on how to fully handle the different use cases. It could be used as a part of a guide in SovereignCloudStack/docs illustrating how to ensure that the standard is adhered to and how to handle setting up new rules in conformance with standard.

Reply:
done so with the creation of the docs PR: https://github.com/SovereignCloudStack/docs/pull/167